### ISSUE TITLE
PowerView API // Shade capabilities table

### ISSUE THREAD

**Issue by andrewfg** [sander76/aio-powerview-api#16]
I am a code contributor for the **_openHAB_** binding for Hunter Douglas PowerView shades (see info below).
- [User Documentation](https://www.openhab.org/addons/bindings/hdpowerview/#hunter-douglas-luxaflex-powerview-binding)
- [Binding Code on GitHub](https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.hdpowerview)
 
Just as you guys have, we are also facing the issue of how to fine tune the binding functionality and UI to match the physical capabilities of the different types of Hunter Douglas shades. As a result of extensive Googling of shade JSON payloads, and developer repositories on GitHub (including yours), I was able to build the following (tentative) table of shade `type` attributes versus `capabilities` attributes. And I am pleased to (hereby) share it with you.

=> Any further inputs or corrections would be very much appreciated - especially from @sander76 and (perhaps) @kingy444 

![image](https://user-images.githubusercontent.com/893994/143322491-63f9ad33-6936-480c-8608-b80edd39f687.png)

**Comment by sander76** [comment]
That's looking good ! Many thanks for this. Also your repo looks very well documented. Better than mine ! ;-)

I used to be working on the European Powerview team as a technical product manager. That's why it was easy for me to get a lot of info of the internals of the system. Unfortunately for this project I am no longer involved in PowerView.

Going through the list see my remarks below:

Silhouette: Can move and tilt-at-bottom
Twist (or double roller): Can move and tilt-at-bottom (doesn't really tilt though, but has 2 pieces of fabric with bands of transparent and non-transparent fabric which run in front of each other and depending on position opens or blocks the view.)
The vertical products (Vertial Slats Left/Right/Split stack): Can move and tilt-anywhere (180)
Curtain (left/right/split) stack : move only
I am not sure at which product "Vertical Tilt 180" refers to. Maybe a duplicate of one of the other vertical products ?

That's about it I guess.
Sander.

**Comment by andrewfg** [comment]
^
Many thanks for the information. If I get any updates from others, then I will post them here too.

**Comment by andrewfg** [comment]
PS I got the `capabilities` descriptions [here](https://github.com/bujvary/hubitat/blob/master/drivers/hunter-douglas-powerview-shade.groovy#:~:text=0%3A%20%22Bottom%20Up,and%20Tilt%2090%C2%B0%22)

**Comment by andrewfg** [comment]
Just for info: here are the Shade Types and Shade Capabilities in code. => If anyone has additions of correc6tions to this list, please feel free to post the information here!

![image](https://user-images.githubusercontent.com/893994/146191336-d3003f8c-4854-466d-bc24-61c570724bd0.png)

![image](https://user-images.githubusercontent.com/893994/146191363-72511ece-87bb-4270-b92b-b864ea010ddf.png)

**Comment by andrewfg** [comment]
> "Silhouette" can also be type 18

@pessorrusso many thanks for the feedback; I will update the table above.

**Comment by carpaij** [comment]
@andrewfg The capabilities of type 6 'Duette' is indeed 0. It's a regular bottom up behaving screen.

**Comment by andrewfg** [comment]
@carpaij many thanks for the confirmation.

**Comment by andrewfg** [comment]
@nixpanic many thanks for the feedback.

> as mentioned in the home-assistant community

Concerning your request on that home-assistant thread.. Unfortunately I cannot solve your request directly. However I can tell you that on **openHAB** we have already implemented support for dual motor shades (i.e. those with independent rail position and tilt actions, or those with top down and bottom up rail positions). And if any of the home-assistant developers want to "borrow" from that code, they can see it [**HERE**.](https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.hdpowerview)

> might be the same as Type 62?

Indeed. I had already picked up type 51 from that thread. You will be able to see the latest version of my ShadeCapabilitiesDatabase starting at line 59 in the source code below..

### [openHAB Shade Capabilities Database](https://github.com/andrewfg/openhab-addons/blob/hdpowerview-shade-capabilities-database/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java)

**Comment by sander76** [comment]
> 

@nixpanic Just checking: they can move and tilt independently from each other ?

**Comment by nixpanic** [comment]
> @nixpanic Just checking: they can move and tilt independently from each other ?

Yes, attached to the ceiling:

* the bottom of the shade can move down/up
* at any up/down position, it is possible to tilt in any direction

**Comment by jeffdeal** [comment]
I have Hunter Douglas "Vignette" Modern Roman Shades with Duolite that report as type 65, and I don't see that in your table. 
 These shades are essentially an inner and outer shade on the same roller.  When both shades are down a room-darkening panel blacks out the room.  When the blackout shade is raised, light filters through the Roman shade. 
 
 If you defined 100% as completely closed and 0% as completely open, 75% would have the room darkening shade halfway up, 50% would be the darkening panel all the way up but Roman shade down, 25% would be the Roman shade halfway up.

I also have Silhouette shades that report as type 23.  These raise and lower and can tilt when all the way closed.

I'd be happy to help improve this integration if I can.

**Comment by andrewfg** [comment]
@jeffdeal many thanks for the feedback. I think we already have the type 23 shades covered. But the type 65 is an interesting new one. And I have some questions about it, so that we can understand more fully.
1. From your description it sounds like the shade has two motorised rails. With fabric extending from the top of the window to the rail at the bottom. Is that correct?
2. On your remote control, is it possible to operate both rails fully independantly? i.e. is it possible to have the back rail fully up and the front rail fully down AND vice versa? Or alternatively are the rails cross linked so that the back rail cannot move down until the front rail is already fully down, OR vice versa?
3. And/or is there some range limitation, whereby (say) raising the front rail causes the back rail to move up when (say) the front rail would start to rise above the back rail (or vice versa).
4. Do you know what `capabilities` value, the shade reports?

**Comment by jeffdeal** [comment]
1.  The Vignette Duolite shades are two shades on one roller.  The front shade and back panel cannot operate independently.  The back panel will only descend once the front shade is all the way down.
2. Since there is only a single roller, they cannot be operated independently.
3. See this link for a more thorough description of the Vignette Duolite:  [https://fredericksburgwindowdecor.com/wp-content/uploads/2020/06/HD_Vignette_Brochure.pdf](url)
4. I'm not sure how to find the capabilities value you reference.  The best way I can describe this shade is a bottom-up shade that has two separate panels.  This site shows a video of the shade operation:  [https://www.youtube.com/watch?v=Mh27XEQ89BQ](url)

Thanks!

**Comment by kingy444** [comment]
@jeffdeal do these currently work on HomeAssistant - I don’t recall code well enough to remember if they would or not but from your description they should function as a normal shade all given with some strange positioning based on the percentage value of the blind

I did think if they don’t match a defined type they default to bottom up but could be wrong.
I would also assume capability is 0 meaning standard bottom up blind anyway

are you just asking for them to work in general or are you wanting specific functionality? Given there is only one motor and they can’t function independently not sure what we could do there

**Comment by andrewfg** [comment]
> not sure how to find the capabilities value 

@jeffdeal could you post the response when you open the link http://<hub-ip-address>/api/shades in a browser?

**Comment by andrewfg** [comment]
I added type 65 to my [openHAB Shade & Capabilities Database](https://github.com/andrewfg/openhab-addons/blob/hdpowerview-shade-capabilities-database/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java) with the provisional `capabilities:0` value; but @jeffdeal it would be really great if you could confirm (or deny) that.

**Comment by andrewfg** [comment]
^
Actually, looking at the capabilities table, I think that yours will have `capabilities:8` but I would still appreciate the confirmation..

![image](https://user-images.githubusercontent.com/893994/150579132-6eaf1b8f-5214-4fb3-af3d-bb268c153a6d.png)

**Comment by jeffdeal** [comment]
I believe 8 will be correct for this shade - lift only Duolite operation.  Thanks!

**Comment by andrewfg** [comment]
> If you defined 100% as completely closed and 0% as completely open, 75% would have the room darkening shade halfway up, 50% would be the darkening panel all the way up but Roman shade down, 25% would be the Roman shade halfway up.

@jeffdeal many thanks for the info about the capabilities. Do you also confirm that the above is an exact description of the rail positions? Do you therefore have any thoughts about how one would model this behaviour in a UI in an ideal world? For instance, perhaps having two sliders where the first one controls the front shade (UI slider 0..100% maps to raw position 0..50%), and the second slider controls the blackout (UI slider 0..100% maps to raw position 50..100%), and it’s operation is disabled and it’s position displays as “N/A” whenever the position of the first UI slider is less than 100%. Or something like that?

**Comment by jeffdeal** [comment]
Hi, I was just using the percentage descriptions to try to describe the way these shades work.  In the official Power View app, the UI shows the shade and the back panel as separate entities. But as previously described, they cannot operate independently.  The shade has to be all the way down before the back panel will come down.

For example, if I have the shade halfway up, by definition the back panel must be all the way up.  In the UI, if I ask to put the back panel halfway down, the shade has to close first, then the back panel will come down to the requested position.  If HA can replicate that functionality it would be consistent with the official app.

The Silhouette shades (type 23) are shown similarly in the UI.  On the left side is the shade position, and the right side are the vanes.  The shade must be all the way down to operate the vanes.  If the shade is down and the vanes are open, and you want to have the shade 50% up the vanes have to close first, then the shade will travel to the requested position.   I'll try to attach a screen shot of the way the shades show in the app:

![Hunter Douglas Duolite](https://user-images.githubusercontent.com/72394729/150648024-a23dbda5-63bf-4b26-a2df-261302ecc211.jpg)

**Comment by andrewfg** [comment]
@jeffdeal thanks for the screen shots. Does it work at all in HA at the moment? And if so, what do you actually see in HA in the following cases?
1. Front shade fully up
2. Front shade half down
3. Front shade fully down, back shade fully up
4. Back shade half down
5. Back shade fully down

**Comment by andrewfg** [comment]
@jeffdeal as a better alternative (in particular if it does not work for you at all in HA at the moment), can you please post the results of moving the shades to each of the 5 cases mentioned above, and then loading this url `http://[hub-ip-address]/api/shades/[shade-id]` in your bowser? To be specific I would like to see the 'positions' fragment, as in the example below, in its entirety i.e. `position1`, `posKind1` -- plus also `position2`, `posKind2` if they are present.

```
{
  "shade": {
    "id": [shade-id],
    "type": [65],
    "capabilities": [8],
    "batteryKind": 2,
    "smartPowerSupply": {
      "status": 0,
      "id": 0,
      "port": 0
    },
    "batteryStatus": 3,
    "batteryStrength": 170,
    "roomId": 32072,
    "firmware": {
      "revision": 1,
      "subRevision": 8,
      "build": 1944
    },
    "motor": {
      "revision": 51,
      "subRevision": 51,
      "build": 11825
    },
    "name": "U2hhZGUgMg==",
    "positions": {
      "posKind1": 1,
      "position1": 65534
    },
    "signalStrength": 4,
    "groupId": 39466
  }
}
```

**Comment by jeffdeal** [comment]
Yes, these shades show up in HA and I can do some rudimentary control like open and close.  There is an attribute called "position" that varies from 0 to 100 where 0 is fully closed and 100 is fully open.  There are up, down, and stop controls available, plus a slider that shows and can control the position.

However, this has no effect on the back panel.  Home Assistant seems unaware of the position of the back panel.  Setting position to "closed" only closes the front shade.

Here's what the entity card of Home Assistant shows when I click on it:


![Vignette in Home Assistant](https://user-images.githubusercontent.com/72394729/150654621-c97ea4f8-555c-485e-acc1-cd4476d71b79.png)

**Comment by andrewfg** [comment]
@jeffdeal that is really interesting; but it opens more questions than giving answrs; so I think it would be very interesting to see the 'positions' values of the five cases mentioned above.

**Comment by jeffdeal** [comment]
Thanks for providing the method of finding the position values.  I had missed that before I sent my previous response.

Here is what the shade reports from the hub at each of the requested positions.  These are the Vignette Duolite shades, type 65:

Front shade fully up:  position1: 65535, posKind1: 1
Front shade half down:  position1: 32339, posKind1: 1
Front shade down, back up: position1: 0, posKind1: 1
Back shade half down:  position1: 33464, posKind1: 2
Back shade fully down: position1: 0, posKind1: 2

ALSO - if you start with the shade and back panel fully down, and raise only the back panel, you get the following:

position1: 65535, posKind1: 2  (equivalent to position1: 0, posKind1: 1)

It's apparent that if posKind1 is 2, the position1 value is referring to the back panel, and if posKind 1 is 1, position1 is referring to the front shade.  Position values are ranging from 0 to 65535, with 0 being closed and 65535 being open.

The Silhouette shades with tilt (type 23) are following the same type of pattern.  When posKind1 is 1, the position1 value refers to the vertical position of the shade, where 0 is closed and 65535 is open.  Tilt is a little different, though:  for tilt the posKind1 value is 3, and the position1 values go from 0 (closed) to 32767 (open).

Thanks again!

Jeff




________________________________

**Comment by andrewfg** [comment]
@jeffdeal That's perfect!

As you say, the functionality of type:65 capabilities:8 shade seems to be analogous to the regular 'bottom up / tilt when closed' shades when the shade is fully down, except that
1. At this point the posKind value is 2 (secondary shade) instead of 3 (tilt / vane), and 
2. The maximum position value of the secondary shade is 64k rather than 32k (90 degree tilt vs 180 degree)

Many thanks for your support!

**Comment by andrewfg** [comment]
BTW I am working on the assumption that the above logic must apply also to shades having `capabilities:9` ('Duolite Lift and Tilt') i.e. shades with **BOTH** a blackout shade **AND** a tilt anywhere function -- see [video](https://www.youtube.com/watch?v=GRQ04HRhTDQ)..

```
    private static final Map<Integer, Capabilities> CAPABILITIES_DATABASE = Arrays.asList(
    // @formatter:off
            new Capabilities(0).primary().tiltOnClosed()                      .text("Bottom Up"),
            new Capabilities(1).primary().tiltAnywhere()                      .text("Bottom Up Tilt 90°"),
            new Capabilities(2).primary().tiltAnywhere().tilt180()            .text("Bottom Up Tilt 180°"),
            new Capabilities(3).primary().tiltOnClosed()                      .text("Vertical"),
            new Capabilities(4).primary().tiltAnywhere().tilt180()            .text("Vertical Tilt 180°"),
            new Capabilities(5)          .tiltAnywhere().tilt180()            .text("Tilt Only 180°"),
            new Capabilities(6).primary()                                     .text("Top Down")                 .primaryStateInverted(),
            new Capabilities(7).primary()                         .secondary().text("Top Down Bottom Up"),
            new Capabilities(8).primary()                                     .text("Duolite Lift")             .withBlackoutShade(),
            new Capabilities(9).primary().tiltAnywhere()                      .text("Duolite Lift and Tilt 90°").withBlackoutShade(),
    // @formatter:on
            new Capabilities()).stream().collect(Collectors.toMap(Capabilities::getValue, Function.identity()));
```

**Comment by trullock** [comment]
I have Silhouette (type 23) blinds which support an arbitrary tilt angle when "closed".

in Home Assistant using the built-in integration which I believe uses this library, I have no control over the tilt (angle or open/close)

Am I missing something, or is this not implemented?

I can see this:

```
class Silhouette(ShadeBottomUpTilt):
    shade_types = (shade_type(23, "Silhouette"),)

    async def tilt_close(self):
        return await self.move({ATTR_POSKIND1: 3, ATTR_POSITION1: MIN_POSITION})

    async def tilt_open(self):
        return await self.move({ATTR_POSKIND1: 3, ATTR_POSITION1: 32767})
```

in `resources/shades.py` but thats about it.

Happy to contribute and test, if I can get some pointers on what the current expected state of things are, or if I'm just being an idiot and missing the controls in HA somehow...

Thanks!

**Comment by andrewfg** [comment]
> Silhouette (type 23) blinds which support an arbitrary tilt angle when "closed".

Surely these shades should support tilt at any position, (not just when closed). Or??

**Comment by pessorrusso** [comment]
> > Silhouette (type 23) blinds which support an arbitrary tilt angle when "closed".
> 
> Surely these shades should support tilt at any position, (not just when closed). Or??

You can adjust the tilt only when fully closed. If you adjust the tilt while it is open the shade will automatically close before being able to tilt.

**Comment by trullock** [comment]
> > Silhouette (type 23) blinds which support an arbitrary tilt angle when "closed".
> 
> Surely these shades should support tilt at any position, (not just when closed). Or??

what @pessorrusso  said.


Is there anything I can do to advance this fix? Let me know how to be helpful 👍

**Comment by sander76** [comment]
> > > Silhouette (type 23) blinds which support an arbitrary tilt angle when "closed".
> > 
> > Surely these shades should support tilt at any position, (not just when closed). Or??
> 
> what @pessorrusso  said.
> 
> 
> Is there anything I can do to advance this fix? Let me know how to be helpful 👍 

Silhouette can only tilt at the bottom. It's the way they are constructed.

**Comment by andrewfg** [comment]
^
So I am a bit confused: the type:23 shades have capabilities:1 which would imply some different functionality than the base capabilities:0 shades; but you guys are reporting that in reality they have the same tilt on closed functionality, as the base shades. Or?

**Comment by trullock** [comment]
My JSON reports `"capabilities": 1`
The whole "blind" can be opened and closed to any arbitrary height.
When its fully down, you can then adjust the tilt of the vanes by an arbitrary amount.
You cannot adjust the vane tilt in any other position other than fully down.

I can't comment on any other models, as these are the ones I have :)

**Comment by andrewfg** [comment]
^
Indeed. The behavior is identical to my shades with capabilities:0; Most peculiar..

**Comment by andrewfg** [comment]
Hmm. What I am seeing from my side, is that it **_SHOULD_** already be working for you; that is why I am confused.

**Comment by trullock** [comment]
Maybe I'm being stupid? How do I double check (humor me)

Adding the "cover" to a dashboard or looking via the integrations/devices/entities pages, i only get up/stop/down buttons...

**Comment by jeffdeal** [comment]
Same here.  And in my case, the up/down/stop buttons and reported position values don't always reflect reality.  For example, I could have a shade all the way up, but the UI the "down" arrow to close the shade won't be available.  I have to hit the up or stop button before the UI enables me to select down.

________________________________

**Comment by andrewfg** [comment]
^
The thing is .. I have to be honest with you guys .. I don't actually use HA on my system, so I cannot physically look at covers or their buttons. The only thing I can look at is the code, which tells me that the normal HDPowerView 'tiltOnClosed' functionality is implemented in the HA codebase. Could it be that you guys are running an older version of the code?

=> Perhaps @sander76 can advise on this?

**Comment by andrewfg** [comment]
^
Hmm. Perhaps this code .. 

https://github.com/sander76/aio-powerview-api/blob/89711e2a0cb4640eb458767d289dcfa3acafb10f/aiopvapi/helpers/shade_types.py#L59

.. should be updated to this ??

```
ATTR_TYPES: (44, 23)
```

**Comment by trullock** [comment]
Githubs 500ing for me at the mo, hoping this email reply works....

I'm busy for a few days but can debug next week. Do we have a confirmed
working in HA blind type I can compare it to?

**Comment by andrewfg** [comment]
> Do we have a confirmed
working in HA blind type I can compare it to?

It seems to be the same as type 44

**Comment by andrewfg** [comment]
@trullock @jeffdeal can you please advise when you operate the tilt via the remote control, is the range of tilt 180 degrees (closed / open/ closed) or just 90 degrees (open / closed)?

**Comment by trullock** [comment]
> @trullock @jeffdeal can you please advise when you operate the tilt via the remote control, is the range of tilt 180 degrees (closed / open/ closed) or just 90 degrees (open / closed)?

It's ~90

**Comment by andrewfg** [comment]
Ok, it sounds as though it should have the same functionality as my Type 44 Twist shades..

**Comment by jeffdeal** [comment]
Tilt on my shades goes between 0 and 90 degrees.
________________________________

**Comment by andrewfg** [comment]
Ok thanks. I think that the code change that I suggested above might fix it for you both..

**Comment by jeffdeal** [comment]
Tilt on my shades is 90 degrees - open/closed.

Jeff
________________________________

**Comment by trullock** [comment]
Does this mean we can publish a fix to make these appear properly? ;)

**Comment by kingy444** [comment]
> @nixpanic many thanks for the feedback.
> 
> > as mentioned in the home-assistant community
> 
> Concerning your request on that home-assistant thread.. Unfortunately I cannot solve your request directly. However I can tell you that on **openHAB** we have already implemented support for dual motor shades (i.e. those with independent rail position and tilt actions, or those with top down and bottom up rail positions). And if any of the home-assistant developers want to "borrow" from that code, they can see it [**HERE**.](https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.hdpowerview)
> 
> > might be the same as Type 62?
> 
> Indeed. I had already picked up type 51 from that thread. You will be able to see the latest version of my ShadeCapabilitiesDatabase starting at line 59 in the source code below..
> 
> ### [openHAB Shade Capabilities Database](https://github.com/andrewfg/openhab-addons/blob/hdpowerview-shade-capabilities-database/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java)

Not sure why the link broke but here is an up to date one

Where have we landed with the new shade types and adding them to this API ? I will look at adding them to HA from there

https://github.com/andrewfg/openhab-addons/blob/main/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

**Comment by andrewfg** [comment]
> Not sure why the link broke 

My fault. When my branch of my fork of the OpenHAB repo was merged into openhab/main, I deleted that branch. And BTW your link above is also on my fork of the repo, so it might suffer the same fate in the future; (although it is currently up-to-date with openhab/main)

=> The safest live link to `ShadeCapabilitiesDatabase.java` is as follows (it will always have the latest version of that file)..

https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

**Comment by trullock** [comment]
From a casual read of the code, I can't see why Sillhouette (type 23) don't already support tilt on close, 

https://github.com/sander76/aio-powerview-api/blob/89711e2a0cb4640eb458767d289dcfa3acafb10f/aiopvapi/resources/shade.py#L221

I'll try and debug it later, perhaps @kingy444 you can start from the other end (HA) and see if you can see whats wrong?

**Comment by kingy444** [comment]
> 

I think you’re right - I didn’t doubt the 23 shades working from HA - my question was aimed at the type 51 which we need to add to the shade types to be detected properly.

I see in the capability database it’s labelled as Venetian blind for openhab- we use these names within HA to describe the blind

**Comment by kingy444** [comment]
> well, 23's tilt function isnt exposed in HA - this is my problem
> 
> _some time later_
> 
> Looks like tilt isnt event supported by ~this lib~ the Hunter Dougless integration? I'm going to have a hack and submit a PR or retract this when I discover what else is wrong 🙈

Correct - I am already submitting a PR for tdbu support for HA support and chatting here to discuss extending that to support tilt 😉 so it’s in train

type 23 shouldn’t be a problem and from
There neither will the others but just wanted to find out how we were handling type 23

don’t think there is anything needed from you in the HA space at the moment @trullock 😉

**Comment by andrewfg** [comment]
> well, 23's tilt function isnt exposed in HA - this is my problem
> type 23 shouldn’t be a problem and from

Did you guys read my prior post?
Your current code says `ATTR_TYPES: (44)` and I think the list should simply be extended => `ATTR_TYPES: (44, 23)`

https://github.com/sander76/aio-powerview-api/issues/16#issuecomment-1028273819

**Comment by kingy444** [comment]
> > well, 23's tilt function isnt exposed in HA - this is my problem
> > type 23 shouldn’t be a problem and from
> 
> Did you guys read my prior post? Your current code says `ATTR_TYPES: (44)` and I think the list should simply be extended => `ATTR_TYPES: (44, 23)`
> 
> [#16 (comment)](https://github.com/sander76/aio-powerview-api/issues/16#issuecomment-1028273819)

I think that whole file appears to be commented out? Unless I’m missing something there - I’m no python expert but I know my way around object oriented languages

type 23 is defined here though under its own class
https://github.com/sander76/aio-powerview-api/blob/89711e2a0cb4640eb458767d289dcfa3acafb10f/aiopvapi/resources/shade.py#L222

**Comment by kingy444** [comment]
> "Silhouette" can also be type 18
> 
> example: `{"id":11895,"type":18,"capabilities":1,"batteryKind":2,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":3,"batteryStrength":162,"roomId":18036,"firmware":{"revision":2,"subRevision":2,"build":2767,"index":9},"name":"TWFzdGVyIDE=","groupId":9889,"positions":{"position1":0,"posKind1":3}},`

Do these also tilt like type 23? I will update and submit a PR so the HA side works if so

**Comment by trullock** [comment]
@kingy444 I can only speak for type 23, capabiltities: 1 (see my posts above)

Happy to test your PR, link me a branch

**Comment by kingy444** [comment]
anyone not on the HomeAssitstant thread i have a Google Drive version pushed out atm (havent commited anything to GIT atm) https://community.home-assistant.io/t/adding-tilt-to-the-hunter-douglas-powerview-cover-integration-luxaflex/233720/35?u=kingy444

Added the first iteration of tilt today

**Comment by trullock** [comment]
> anyone not on the HomeAssitstant thread i have a Google Drive version pushed out atm (havent commited anything to GIT atm) https://community.home-assistant.io/t/adding-tilt-to-the-hunter-douglas-powerview-cover-integration-luxaflex/233720/35?u=kingy444
> 
> Added the first iteration of tilt today

Awesome, I was going to ~nag~ follow up about this today ;) I'll help test!

**Comment by kingy444** [comment]
> @pessorrusso can you confirm re type 18

@pessorrusso new post on HA forum - can you please test type 18

**Comment by kingy444** [comment]
> Just for info: here are the Shade Types and Shade Capabilities in code. => If anyone has additions of correc6tions to this list, please feel free to post the information here!
> 
> ![image](https://user-images.githubusercontent.com/893994/146191336-d3003f8c-4854-466d-bc24-61c570724bd0.png)
> 
> ![image](https://user-images.githubusercontent.com/893994/146191363-72511ece-87bb-4270-b92b-b864ea010ddf.png)

Hey - New shade from HA forum - could you give me a hand understanding the capabilities too
Like for instance - how does capability 2 and 4 differ ?

how does type 0 differ from type 3 (we have shades in the same class here under ShadeBottomUp.

 Then our class ShadeBottomUpTiltAnywhere has capability 2+3 listed together too - but from your code i would think perhaps this is wrong as cabability 3 is only tiltOnClosed and 2 can tiltAnywhere

maybe a breakdown of what your definitions mean ?

`  "shadeData": [
        {
            "id": 48556,
            "type": 38,
            "capabilities": 9,
            "batteryKind": 2,
            "smartPowerSupply": {
                "status": 0,
                "id": 0,
                "port": 0
            },
            "batteryStatus": 3,
            "batteryStrength": 177,
            "roomId": 49569,
            "firmware": {
                "revision": 2,
                "subRevision": 1,
                "build": 2757,
                "index": 20
            },
            "name": "TWFzdGVyIEJlZHJvb20gMQ==",
            "positions": {
                "position1": 32713,
                "posKind1": 3
            },
            "groupId": 23871
        },
        {
            "id": 19192,
            "type": 23,
            "capabilities": 1,
            "batteryKind": 2,
            "smartPowerSupply": {
                "status": 0,
                "id": 0,
                "port": 0
            },
            "batteryStatus": 3,
            "batteryStrength": 181,
            "roomId": 9189,
            "firmware": {
                "revision": 2,
                "subRevision": 1,
                "build": 2690,
                "index": 12
            },
            "name": "TWFzdGVyIEJhdGhyb29tIDI=",
            "groupId": 19501,
            "positions": {
                "posKind1": 1,
                "position1": 0
            }
        },`

**Comment by andrewfg** [comment]
> how does capability 2 and 4 differ ?
> how does type 0 differ from type 3 ?

I believe 0 & 2 move the shades in the Y axis (up/down), whereas 3 & 4 move them in the X axis (left/right). but apart from the axis orientation, the control actions resp. API command/responses should be the same.

> Then our class ShadeBottomUpTiltAnywhere has capability 2+3 listed together too - but from your code i would think perhaps this is wrong as cabability 3 is only tiltOnClosed and 2 can tiltAnywhere

I can't really answer this because I don't own any such shades. However as a general rule, I think there should be a separate class to implement each capability type. Albeit as noted above the X axis and Y axis flavours of the capabilities could have the same implementation.

> Am i correct in assuming if a shade has a certain capability their API JSON will be the same?

Yes I think so. (Except obviously for the exact model/instance/user specific data).

> Then we have type 44 which we have with a class of ShadeBottomUpTilt but based on capabilities this is the same capability of a M25T Roller Blind which definitely doesnt have tilt

You are probably right. I have some type 44 shades so I can test these *) and can confirm the bottom up tilt functionality. I don't have an M25T to test, but from what you say, it sounds as though it is wrongly classified as capabilities 0; -- maybe it should be capabilities 8 ??

*) For info type 44 has a loop of fabric on a pulley with alternating light and dark strips, so when the dark strips on one side overlap the light strips on the other, the overall result is the darkest. This is actually implemented with one shade motor, so it can actually be light and dark in any position. However when the shades are fully down then the tilt command can be used to adjust (fine tune) the overlap.

> Type 4 capability - is there any difference here on type 1 ? Get one is vertical and one is bottom up but i assume the API call is the same?

Yes. See my comment above concerning 2 & 4 resp. 0 & 3. But I don't think 1 & 4 are equivalent.

> Any thoughts on the API calls needed for capability type 8/9

I will post a second message on this topic.

**Comment by pessorrusso** [comment]
> > @pessorrusso can you confirm re type 18
> 
> @pessorrusso new post on HA forum - can you please test type 18

I'll be able to test this weekend.

**Comment by andrewfg** [comment]
> do these one respectively treat the primary / secondary shades MAX value as 0 and MIN as 65K?

Yes. I think these shades have bottom rail fixed to the floor, motor at the top, that lifts top rail up from the floor. So when the motor is down, the shade is open.

**Comment by andrewfg** [comment]
> Damn thats annoying - I have the M25T and it definitely has capability 0

Yeah. I guess that when HD invented the system, they only had single motor (position1) shades, and capability 0 was the first. So when they realised they could do tilt and vane stuff (e.g. venetian blinds) without needing a second (position2) motor, simply by driving the position1 motor a bit further beyond its closed position, they probably added that to capability 0 as a kind of freebie extra. But did not want to change the firmware in legacy simple shades (like your M25T) to indicate that those did not have the extra tweaks. Or something like that.

=> Just out of interest, does anything happen on the M25T when you press the tilt/vane buttons on your HD remote?

**Comment by kingy444** [comment]
> > `"type": 38, "capabilities": 9`
> 
> @kingy444 Can you please let me know the model name for your type 38 / caps 9 shade? And what it looks like?

Silhouette Duolite apparantly

**Comment by kingy444** [comment]
> > > `"type": 38, "capabilities": 9`
> > 
> > 
> > @kingy444 Can you please let me know the model name for your type 38 / caps 9 shade? And what it looks like?
> 
> Silhouette Duolite apparantly

that will give us 1 of each capability except for type 4 i believe

**Comment by andrewfg** [comment]
PS @kingy444 you may be interested in the 'HD Powerview API Appendix' created by @jlaur. (I suggest you contact him in case updates are available..)

[HD Powerview API Appendix.pdf](https://github.com/sander76/aio-powerview-api/files/8717762/HD.Powerview.API.Appendix.pdf)

**Comment by andrewfg** [comment]
@kingy444 I also saw someone was mentioning a type 18 shade (another new one for me) =>do you have any JSON info for that one yet?

**Comment by kingy444** [comment]
> @kingy444 I also saw someone was mentioning a type 18 shade (another new one for me) =>do you have any JSON info for that one yet?

They appear follow the same format as type 23 (and are named the same)

posted here already actualy ;)
https://github.com/sander76/aio-powerview-api/issues/16#issuecomment-1011157777

**Comment by andrewfg** [comment]
@kingy444 many thanks. Below are my two tables updated to state of the art with types 18 and 38..

![image](https://user-images.githubusercontent.com/893994/169089574-2f87877d-6de1-4fcf-8e0e-6d98792fe680.png)

![image](https://user-images.githubusercontent.com/893994/169089726-58b318c2-9bde-4617-86e7-bc7a08ab1820.png)

**Comment by kingy444** [comment]
@andrewfg 

> @jeffdeal That's perfect!
> 
> As you say, the functionality of type:65 capabilities:8 shade seems to be analogous to the regular 'bottom up / tilt when closed' shades when the shade is fully down, except that
> 
> 1. At this point the posKind value is 2 (secondary shade) instead of 3 (tilt / vane), and
> 2. The maximum position value of the secondary shade is 64k rather than 32k (90 degree tilt vs 180 degree)
> 
> Many thanks for your support!

I still need to have a look at your code (been a while since I used Java) but do you know offhand if these type 8 capability accept a json comma on containing both the position1 and position2 in 1?

we send a close command by default of poskind1=0 but for top down/bottom up we send both positions as 0 in the same command

@jeffdeal if you could close your shade and then check the result of http://<powerviewip>/api/shades/<shadid>?refresh=true that could answer the question too

I really hate that HD have used position 1 for both blind position and tilt position on some

It makes keeping shade position internally in HA difficult

**Comment by jeffdeal** [comment]
Hi folks, here's the result of the call against my hub.  I have 6 shades; 3 of type 23 and 3 of type 65.  All shades were closed when this was collected.  Let me know if you need something else!

{"shadeIds":[19733,53613,10539,9040,53946,45588],"shadeData":[{"id":19733,"type":23,"capabilities":1,"batteryKind":2,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":3,"batteryStrength":188,"roomId":9155,"firmware":{"revision":2,"subRevision":1,"build":2690,"index":12},"name":"TGl2aW5nIFdlc3Q=","groupId":50255,"positions":{"posKind1":1,"position1":0}},{"id":53613,"type":23,"capabilities":1,"batteryKind":2,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":0,"batteryStrength":0,"roomId":9155,"firmware":{"revision":2,"subRevision":1,"build":2690,"index":12},"name":"TGl2aW5nIFNvdXRo","groupId":50255,"positions":{"posKind1":1,"position1":123}},{"id":10539,"type":65,"capabilities":8,"batteryKind":2,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":3,"batteryStrength":188,"roomId":2356,"name":"TWVsaXNzYQ==","firmware":{"revision":2,"subRevision":0,"build":1979,"index":48},"groupId":58492,"positions":{"posKind1":2,"position1":0}},{"id":9040,"type":65,"capabilities":8,"batteryKind":2,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":0,"batteryStrength":0,"roomId":45897,"firmware":{"revision":2,"subRevision":0,"build":1979,"index":48},"name":"TWFzdGVy","groupId":64862,"positions":{"posKind1":2,"position1":0}},{"id":53946,"type":23,"capabilities":1,"batteryKind":2,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":3,"batteryStrength":155,"roomId":49316,"firmware":{"revision":2,"subRevision":1,"build":2690,"index":12},"name":"RGluaW5n","groupId":41457,"positions":{"posKind1":1,"position1":0}},{"id":45588,"type":65,"capabilities":8,"batteryStatus":0,"batteryStrength":0,"roomId":45351,"firmware":{"revision":2,"subRevision":0,"build":1979,"index":48},"name":"SmFrZQ==","groupId":59977,"smartPowerSupply":{"status":0,"id":0,"port":0},"positions":{"position1":0,"posKind1":2},"signalStrength":0}]}

**Comment by andrewfg** [comment]
> do you know offhand if these type 8 capability accept a json comma on containing both the position1 and position2 in 1?

@kingy444 in my experience you can always send a JSON payload to the hub with both a position1 and a position2 in it. You can even send a secondary position in position1 and a primary position in position2. And the hub will sort it out, and send the correct commands to the shade motors. And it seems the JSON response from the hub will contain the sorted out values. Either with a single position element or two position elements depending on the actual state of the shade and it’s capabilities. It always responds with at least one position element, and it MAY respond with two..

The trick when sending two position elements, is that the values sent must reflect the physical constraints of the given shade type. So, for example 1) you cannot command an upper secondary rail to go below a lower primary rail, nor vice-versa; 2) you cannot command a vane position on a tilt on close shade unless the primary rail is closed; 3) you cannot command a secondary blackout shade position unless the primary shade is closed, nor vice-versa, etc. etc.

**Comment by andrewfg** [comment]
^
Correction to above: if you send a JSON payload to the hub with a munged combination of position1 and position2 values, the hub will respond parrot fashion with what you sent it. However if you do ‘refresh=true’ the hub will respond with the de-munged position(s) as appropriate.

**Comment by kingy444** [comment]
> > do you know offhand if these type 8 capability accept a json comma on containing both the position1 and position2 in 1?
> 
> @kingy444 in my experience you can always send a JSON payload to the hub with both a position1 and a position2 in it. You can even send a secondary position in position1 and a primary position in position2. And the hub will sort it out, and send the correct commands to the shade motors. And it seems the JSON response from the hub will contain the sorted out values. Either with a single position element or two position elements depending on the actual state of the shade and it’s capabilities. It always responds with at least one position element, and it MAY respond with two..
> 
> The trick when sending two position elements, is that the values sent must reflect the physical constraints of the given shade type. So, for example 1) you cannot command an upper secondary rail to go below a lower primary rail, nor vice-versa; 2) you cannot command a vane position on a tilt on close shade unless the primary rail is closed; 3) you cannot command a secondary blackout shade position unless the primary shade is closed, nor vice-versa, etc. etc.

How are you handling type 18/23 shade position when you move the tilt on the UI

> ^
> Correction to above: if you send a JSON payload to the hub with a munged combination of position1 and position2 values, the hub will respond parrot fashion with what you sent it. However if you do ‘refresh=true’ the hub will respond with the de-munged position(s) as appropriate.

Yea I tested this with the tdbu

mainly around the shades that don’t send poskind 1 when they tilt a blind and we just have to assume the position

For others that always present the value we are returning the value directly from the JSON now as this works much more consistently but the shades that only send one position back means we can’t do that for those shades. Bit of refactoring to do there

@sander76 trying to update the base code again which is simple enough but could you maybe provide a little info on how you are running tests locally with NOX?
Ideally I would want to pull my edited code directly into My HomeAssistant DEV docker but haven’t figured that one out either

**Comment by andrewfg** [comment]
> How are you handling type 18/23 shade position when you move the tilt on the UI

Generally a tilt on closed shade will only use one position element in its JSON. If it’s posKind is for the primary shade, and it’s position value is less than fully down, then the vane position is de-facto UNDEF. And if it’s posKind is for the vane position, then the primary rail position is de-facto fully down.

Note that because of the above-mentioned position interlocking, there is one specific state ‘fully down, but not tilted’ that can be achieved/described by either of two possible commands: 1) primary position == fully down, or 2) vane position == not tilted.

**Comment by kingy444** [comment]
> => Just out of interest, does anything happen on the M25T when you press the tilt/vane buttons on your HD remote?

yep - they treat that as a up down command - as do my type 6 - but both of these have capability 0 so that isnt surprising.

**Comment by kingy444** [comment]
> Front shade fully up: position1: 65535, posKind1: 1 Front shade half down: position1: 32339, posKind1: 1 Front shade down, back up: position1: 0, posKind1: 1 Back shade half down: position1: 33464, posKind1: 2 Back shade fully down: position1: 0, posKind1: 2

@jeffdeal as i understand you cannot move these without the other being in a particular position - are you comfortable posting commands to the shades to test or perhaps @andrewfg  has the answer

@andrewfg from the openhab code it looks like you invert poskind=2 for supportsBlackoutShade which is good to know

Really - i just need to work out the JSON that needs to be sent for open position + close position, then how we are to control the shade for position

We could create two entiite like we do for tdbu - if the act independantly then i guess we need the open/close defined for each shade type and two entities in ha

**Comment by andrewfg** [comment]
> if the act independantly then i guess we need the open/close defined for each shade type and two entities in ha

In OH each shade can (potentially) have three channels -- namely the primary shade, the secondary shade, and the vane; and their respective functionality is defined depending on the shade's advertised capabilities, as follows..
- Channels are dis/en-abled depending on whether shade actually has primary/secondary rails and/or vanes.
- Disabled channels return position 'undefined' in UI.
- Adjust max range of vane/tilt channel 0..32k resp. 0..64k depending on whether shade has 90 or 180 degree tilt.
- Invert range of primary/secondary rails depending on whether the rail is bottom up or top down.
- Interlocking between primary position and vane position values on tilt on closed shades.
- Interlocking between primary position and secondary position values on TDBU or Duolite shades.
- Read/write position2/posKind2 (in addition to position1/posKind1) if the shade requires it.
- etc.

> @jeffdeal as i understand you cannot move these without the other being in a particular position .. or perhaps @andrewfg has the answer

I think he already gave that information in his [original post](https://github.com/sander76/aio-powerview-api/issues/16#issuecomment-1019308024)

**Comment by jeffdeal** [comment]
Good day, I'd be happy to test sending commands to shades but would need some instructions on how to do so.

My Silhouette shades are roller shades with tilt, type 23, capabilities 1.  The vanes tilt from 0 to 90 degrees, only when the shade is fully down.

My Vignette Duolite are roller shades with a darkening back panel, type 65, capabilities 8.  Starting from fully open, the light filtering panel lowers first, then the blackout shade lowers next.  The blackout panel will only raise and lower with the light filtering panel fully down.

Jeff

________________________________

**Comment by andrewfg** [comment]
```
Front shade fully up: position1: 65535, posKind1: 1
Front shade half down: position1: 32339, posKind1: 1
Front shade down, back up: position1: 0, posKind1: 1
Back shade half down: position1: 33464, posKind1: 2
Back shade fully down: position1: 0, posKind1: 2
```

@kingy444 if you stick the back (secondary) shade positions in position2/posKind2 then you will always have both the primary shade data (in position2/posKind2) and the secondary shade data (in position1/posKind1)..

**Comment by kingy444** [comment]
> ```
> Front shade fully up: position1: 65535, posKind1: 1
> Front shade half down: position1: 32339, posKind1: 1
> Front shade down, back up: position1: 0, posKind1: 1
> Back shade half down: position1: 33464, posKind1: 2
> Back shade fully down: position1: 0, posKind1: 2
> ```
> 
> 
>     
>  
>   
> 
> @kingy444 if you stick the back (secondary) shade positions in position2/posKind2 then you will always have both the primary shade data (in position1/posKind1) and the secondary shade data (in position2/posKind2)..
> 
> 
> 
> EDIT: for shades with a primary and tilt anywhere capability you stick the primary shade position in position1/posKind1 and the secondary in position2/posKind2..

From the description I was working on the expectation that only position1 was being processed(similar to type 23 - they will accept position2 but ignore it)

@andrewfg do you know from your testing of these if sending a poskind2 command moves poskind1 as well? The way I was reading it was the back shade would move until the front one did or are those completely seperate and operate independently. 
Perhaps for @jeffdeal but could you have the back open 25% and the front open 40% for instance

unfortunately HA architecture restricts to 1 shade and tilt - I did put in a request to add support but it seems to not be on the cards

**Comment by jeffdeal** [comment]
On type 23, the vanes cannot tilt unless the shade is completely closed.

On type 65, the front and back panel are NOT independent, as they are on a single roller.  The front shade must be fully closed before the back panel can be lowered.  Likewise, the back panel must be completely open before the front shade can be raised.

Jeff


________________________________

**Comment by kingy444** [comment]
> On type 65, the front and back panel are NOT independent, as they are on a single roller. The front shade must be fully closed before the back panel can be lowered. Likewise, the back panel must be completely open before the front shade can be raised.

that makes it quite interesting implementing these in HA as moving one relies on the other. @andrewfg from your experience what are the position restrictions on these ? ie similar to type 23 - poskind 1 appears to need to be 65535 before we send and poskind2 above 0? Do you have these as primary and secondary in openhab or a single slider?

To make sure i have it clear in my head lets try to use some consistent terminology when we talk about where the shade is. Lets try open closed rather than up/down as this can get confusing. I have tried to label how i am understanding below but please confrim

Front shade fully up: position1: 65535, posKind1: 1 (front open, rear ?)
Front shade half down: position1: 32339, posKind1: 1 (front halfway, rear closed)
Front shade down, back up: position1: 0, posKind1: 1 (front closed, rear open)
Back shade half down: position1: 33464, posKind1: 2 (front closed, rear halfway)
Back shade fully down: position1: 0, posKind1: 2 (front ?, rear closed)


To run some tests follow the below:

1. Install Advanced REST Client - https://chrome.google.com/webstore/detail/advanced-rest-client/hgmloofddffdnphfgcellkdfbfbjeloo
2. You will want to fill it out so it looks like this (tried to highlight key areas)
![image](https://user-images.githubusercontent.com/7148670/169653352-121fbf45-630d-45bd-b207-5ec264dfa786.png)

From there you can post some different positions and test how the blind acts. @andrewfg suggests you can post both positions at once - so perhaps try something like the below.
You obviously understand how these work so feel free to test some other combinations. The max value you can post is 65535 - ideally posting both positons will move the shade to where you want it but im not entirely sure on the 'clamping' we need to do to prevent sending positions that are impossible

I am hoping we can find a way to implement as two seperate entities, but it will require posting both poskinds at once, or when posting poskind2 to the desired location posikind1 will automatically move to closed for instance (or the equivelent of where it should be) this is how type 21 works.

from fully closed
`{
  "shade": {
    "positions": {
      "posKind1": 1,
      "position1": 65535,
      "posKind2": 2,
      "position2": 33464
    }
  }
}`
from fully closed
`{
  "shade": {
    "positions": {
      "posKind1": 1,
      "position1": 33464,
      "posKind2": 2,
      "position2": 48000
    }
  }
}`
from fuly open
`{
  "shade": {
    "positions": {
      "posKind1": 2,
      "position1": 20000,
    }
  }
}`
from fully open
`{
  "shade": {
    "positions": {
      "posKind1": 1,
      "position1": 20000,
    }
  }
}`

**Comment by andrewfg** [comment]
> @andrewfg do you know from your testing of these if sending a poskind2 command moves poskind1 as well?

I am not completely sure. But in general I think this it could happen. If you send an impossible combination of positions to a shade, it will generally try to do SOMETHING. It obviously cannot achieve the commanded impossible position, but it will move to some position within the range of what is possible. I am not sure whether if (say) commanding a TDBU bottom rail above its top rail would a) move the top rail too, or b) stop when the bottom rail hits the top rail. I could imagine both possible scenarios. Perhaps it might even depend on the firmware.

This is why personally I think the code should filter out impossible combinations and not try to send them to the shade. That way WYSIWYG.

But of course if you don't filter out impossible combinations you can rely on the device to sort itself out, and you will be able to read whatever compromise position it reached when you do the next `refresh=true`

> @kingy444 if you stick the back (secondary) shade positions in position2/posKind2 then you will always have both the primary shade data (in position1/posKind1) and the secondary shade data (in position2/posKind2)..

Actually, on reflection, I think my suggestion above may be WRONG. Sorry for that. From my description [here](https://github.com/jlaur/hdpowerview-doc/issues/1#issuecomment-1133659494) I believe that devices with two dimensions of motion that are tightly physically interlocked in fact put both positions (logically interlocked) in just position1/posKind1. At least that is the way it is coded (and tested as working) in OH. So please ignore my prior post.

**Comment by kingy444** [comment]
What I am now thinking for type 8 at least (I’m not sure on how to implement for type 9 (they are tdbu and tilt right ? @andrewfg ))

@jeffdeal your current description is that the shade works in HA for the front panel 0-100 is representing the front shade all the way open at 100 right. And if I have my logic correct the back shade cannot start moving until this occurs right ? Lack of secondary rail support is issue here as I would need a second entity 

so what if we implemented special logic that 0-50 represented the front shade and 50-100 represented the back shade. So say 75 would be equivalent to front shade fully open back shade half open ?

**Comment by kingy444** [comment]
> This is why personally I think the code should filter out impossible combinations and not try to send them to the shade. That way WYSIWYG.

I definitely agree with you here, and we are doing that for the TDBU but the issue is “what is an impossible position”
TDBU top shade can’t be below bottom shade - pretty simple - just need to work out the logic in laments terms for other impossible positions and translate that to the code equivalent 

So like with these type 8 it sounds like then the front shade is below 65K the back shade cannot be above 0 and likewise then the back is above 0 the front cannot be belo 65
(Did I get that right ?)

**Comment by andrewfg** [comment]
> not sure on how to implement for type 9 (they are tdbu and tilt right ? @andrewfg ))

A caps 9 shade is like a caps 8 but with a tilt motor added.
- Caps 8 and Caps 9: the primary and secondary shade positions are packed using the interlock logic in position1/posKind1
- Additionally Caps 9 only:  the tilt position is independently in position2/posKind2

---
> “what is an impossible position”

From my perspective the following are impossible positions, and how they interlock..
```
// Tilt on closed (a)
if (shadePosition != 0) {
  vanePosition = 0;
}

// Tilt on closed (b)
if (vanePosition != 0) {
  shadePosition = 0;
}

// Tilt shades in general
if (tilt90degrees) {
  vanePosition = Math.Min(vanePosition, 32k};
}

// Top down, Bottom up
bottomPrimaryShadePosition = Math.Min(bottomPrimaryShadePosition, (64K - topSecondaryShadePosition));
topSecondaryShadePosition = Math.Min(topSecondaryShadePosition, (64K - bottomPrimaryShadePosition));

// Duolite (a)
if (primaryShadePosition != 0) {
  secondaryShadePosition = 64k;
}

// Duolite (b)
if (secondaryShadePosition != 0) {
 primaryShadePosition = 0;
}
```

**Comment by kingy444** [comment]
> I'm not quite sure how I should test

Just try moving the shade slider and report how it works. Similar to the REST commands we sent (all the code does is generate the JSON)

So test things like setting to 25 (should open back halfway) then move straight to 75 (should open the back fully and open front halfway)
Pressing close should close both shades and pressing open should open both

test something similar with the type 23 too if you don’t mind

**Comment by kingy444** [comment]
> Sending posKind1 = 2 and position1 between 0 and 65535 will place the back panel in the requested position, with 0 being completely closed (down) and 65535 being up (open).



> Sending posKind1 = 1 always refers to the front panel, with position values of 0 representing down (closed) and 65535 up (open)

@andrewfg sorry for keeping lean on you mate - appreaciate your help

little confused with the comments above - i thought the blackoutshades had secondary inverted (0 open and 65535 closed) and i thought your code actually reflected that
I have coded from my end to align with the reports from @jeffdeal but appreaciate your input as ever

**Comment by kingy444** [comment]
> Just try moving the shade slider and report how it works. Similar to the REST commands we sent (all the code does is generate the JSON)
> 
> So test things like setting to 25 (should open back halfway) then move straight to 75 (should open the back fully and open front halfway) Pressing close should close both shades and pressing open should open both
> 
> test something similar with the type 23 too if you don’t mind

@jeffdeal  all this still applies but can you download the new version - i added the two hidden shades i spoke of - you will need to browse to the device to see them as they are hidden

Go to Intergrations - select devices - select the type 65

![image](https://user-images.githubusercontent.com/7148670/170219586-9aa6e647-17d6-4131-92b1-d8613b40f81d.png)

**Comment by andrewfg** [comment]
@kingy444 

> thought the blackoutshades had secondary inverted (0 open and 65535 closed) and i thought your code actually reflected that

Nope. (Actually in OH ALL bottom up shades are treated as ‘inverted’ since the standard OH UI RollerShutter control considers up (resp. off) to be position 0% and down (resp. on) to be 100%..)

**Comment by jeffdeal** [comment]
Good day, I installed the custom component, configured it, and gave it a try today.  The type 23s work great with the position slider!  The type 65s did not work as described.  Using the slider only commanded the front panel, so any position requested always raised the back panel first.  Setting the shade fully closed only did so for the front panel.

**Comment by kingy444** [comment]
> Good day, I installed the custom component, configured it, and gave it a try today. The type 23s work great with the position slider! The type 65s did not work as described. Using the slider only commanded the front panel, so any position requested always raised the back panel first. Setting the shade fully closed only did so for the front panel.

MMM that doesnt make a lot of sense - i just tested the json and it all looks to be generating correctly. I would expect the 'Front' to act the way you described and the back to do the inverse

Combined should then act the way i described sending 0-50 and 51-100 commands

In my Screenshot you will see there are 3 entitties created now - did you test all 3 ?

I have just popped another version up that will pop warning messages into your log when you move the slider to identify what commands are being sent to the shade. This should help identify what is being sent and the expected postion. Please try to match this with what you provided above as that is what i tried to replicate

You want to check all 3 here. Can you please 

> Go to Intergrations 
![image](https://user-images.githubusercontent.com/7148670/170267937-7df03bd9-4dcb-4fb2-8991-2a16a39e2a45.png)
- select devices 
![image](https://user-images.githubusercontent.com/7148670/170267978-d2cac98a-12de-49b5-a1c0-2c4a5fe3abdb.png)
- select the type 65
![image](https://user-images.githubusercontent.com/7148670/170268049-75d7c77f-759e-450c-aa7b-071df206678d.png)
- Test the 3 shades
> 
> ![image](https://user-images.githubusercontent.com/7148670/170219586-9aa6e647-17d6-4131-92b1-d8613b40f81d.png)

NOTE: If you remove the integration and re-add the 3 are no longer hidden and can be access via main UI

**Comment by jeffdeal** [comment]
I've not been able to see the controls that you are showing.  I've removed and re-added the custom component.  I added the hub, and devices got created with a _2 suffix so they're easy to identify.  I can see my three type 65 shades on BOTH the standard and custom integrations, and can tell them apart.  I have not seen a control panel like you're showing, though.  Here's a screen shot of what's in my custom_components directory to see if I have the right files installed.  

![HD custom](https://user-images.githubusercontent.com/72394729/170290791-a20f7ca2-96c0-4640-8c26-385a9dafad2b.png)

**Comment by kingy444** [comment]
> Here's a screen shot of what's in my custom_components directory to see if I have the right files installed.

Can you re-download the custom component and install it again please - there is another version.txt in there to identify which one you have

Remove the one you currently have (both the custom and core so there are no conflicts) reboot your HA and then readd the integration

If they are recognised as type 65 successfully you will have 3 entities created for each shade

If you wish to get to the control panel try and follow the screenshots I sent above

**Comment by jeffdeal** [comment]
Sorry to be so needy with this testing - I have tried to follow your instructions and can't seem to get to the control for the type 65s.  Here's what I've tried:

- Deleted the existing Hunter Douglas integration from my Integrations settings
- Deleted the custom integration from Integrations, deleted and removed the files from custom_components
- Re-downloaded the custom integration and installed it, restarted HA
- The custom component discovered my hub so I configured it.  It discovered 7 devices (the hub and 3 type 23, 3 type 65 shades.
- I went into devices and picked one of my type 65s, and went to the control section.  I do not see the extra controls.  On the type 23s I can see the separate tilt section.

I am just not sure what I'm doing wrong.  This is all I've ever seen for my type 65s:

![type 65 controls](https://user-images.githubusercontent.com/72394729/170548313-d7ae17ed-b3be-4c9b-aa48-d466caa780fb.png)

**Comment by kingy444** [comment]
> Sorry to be so needy with this testing - I have tried to follow your instructions and can't seem to get to the control for the type 65s. Here's what I've tried:

Not needy mate - side effect of being unable to test completely locally - believe i found the bug. Nothing you were doing wrong

Can you re-download and repeat the process please

**Comment by jeffdeal** [comment]
Whew!  Thanks, I was losing my mind.  I downloaded the new component today and it works as described.  I can now see the separate entities for the combined, front, and back on the type 65 shades, and each is working.

**Comment by kingy444** [comment]
@andrewfg  could you confirm for me - are the type 7, capability 6 shades (Top Down) only referred in that way because of the functionality of the motor (inverted) or are they actually top down - couldnt find much on them online

Know i pinged you on the other forum too but if you have the data for palm beach shutters that would be great

**Comment by andrewfg** [comment]
Apropos the top downs, have a look at the pictures I posted here..

https://github.com/jlaur/hdpowerview-doc/issues/1#issuecomment-1136141254

**Comment by pessorrusso** [comment]
> 
> @pessorrusso just a thought - could the type 18 actually be Pirouette and not Silhouette?
> 
> It is entirely possible they are Silhouette but as the only difference is the shape of the fabric as far as I am aware thought I might ask as we don’t have a definition for Pirouette but we have 2 for Silhouette

:) I just checked the HD website and my receipt, yes it is a Pirouette. Sorry the confusion.
You code is working fine for it.

**Comment by jpearl** [comment]
Adding a few unmentioned shade types. In particular, looking to see what adding support for `33` would be, though i suspect it's identical to the existing TDBU implementation

|type|capabilities|name|comment
|-|-|-|-|
|31|0|vignette|works today with default implementation
|33|7|duette architella|TDBU - does NOT currently work

**Comment by kingy444** [comment]
> Adding a few unmentioned shade types. In particular, looking to see what adding support for `33` would be, though i suspect it's identical to the existing TDBU implementation
> 
> type	capabilities	name	comment
> 31	0	vignette	works today with default implementation
> 33	7	duette architella	TDBU - does NOT currently work

Confirming json is best but these shouldn’t be a problem

someone asked for type 33 today but haven’t heard back from them yet to confirm

likelyhood is type 7 works as is, we are just never confident as there are two implementations of type 0

https://community.home-assistant.io/t/adding-tilt-to-the-hunter-douglas-powerview-cover-integration-luxaflex/233720/195

**Comment by kingy444** [comment]
@andrewfg just for your attention in the openhab space mate 😉

**Comment by jpearl** [comment]
For 33:

(Note: there's a slight chance these are reversed; left commentary in the table)

|position|json|
|-|-|
|Top fully open / bottom fully closed  (shade material is sitting on the bottom of the window)|`"positions":{"posKind2":2,"position2":65535,"posKind1":1,"position1":0}}}`
|Top fully closed / bottom fully open (shade material is all the way at the top) |`"positions":{"posKind2":2,"position2":0,"posKind1":1,"position1":65535}}}`
|Both shades open somewhere around middle (40/60 ish)|`"positions":{"posKind2":2,"position2":18510,"posKind1":1,"position1":21039}}}`

**Comment by kingy444** [comment]
> For 33:
> 
> (Note: there's a slight chance these are reversed; left commentary in the table)
> 
> position	json
> Top fully open / bottom fully closed (shade material is sitting on the bottom of the window)	`"positions":{"posKind2":2,"position2":65535,"posKind1":1,"position1":0}}}`
> Top fully closed / bottom fully open (shade material is all the way at the top)	`"positions":{"posKind2":2,"position2":0,"posKind1":1,"position1":65535}}}`
> Both shades open somewhere around middle (40/60 ish)	`"positions":{"posKind2":2,"position2":18510,"posKind1":1,"position1":21039}}}`

Perfect mate - definitely standard tdbu and inclusion will be simple 😉

**Comment by andrewfg** [comment]
@jpearl thanks for reporting, @kingy444 thanks for confirming

I am just updating the OpenHAB code base to match yours..

https://github.com/openhab/openhab-addons/pull/13110

**Comment by andrewfg** [comment]
@kingy444 do you have any user data on shades type 55 or 56 ? My reason for asking is that I have a user whose shades report as below (his type 56 is basically the same as the 55), and they have vertical vanes with main position / tilt anywhere / tilt 180. 

So I think that we may have the following errors / mix ups..
- capabilities:4 and capabilities:3 should be interchanged
- shade types 54 thru 56 should be capabilities:4
- shade types 69 thru 71 should be capabilities:3

```
{"shade":{"id":20034,"type":55,"capabilities":4,"batteryKind":3,"smartPowerSupply":{"status":0,"id":0,"port":0},"batteryStatus":3,"batteryStrength":146,"roomId":39451,"name":"T2ZmaWNlIFdpbmRvdw==","groupId":39348,"firmware":{"revision":1,"subRevision":8,"build":1944},"motor":{"revision":2,"subRevision":1,"build":0},"positions":{"posKind1":1,"position1":26963,"posKind2":3,"position2":26565},"signalStrength":4}}
```

**Comment by kingy444** [comment]
> * shade types 54 thru 56 should be capabilities:4

definitely have report of both 54 and 55 being capability 3 but the same post also has type 55 as capability 4 -wtaf

Linked and copied important part below

but as you can see they have 4 of these and 2 are reporting capabiity 3 and 2 are reporting capability 4

the two reporting capability 3 have completely different motor revisions
the two reporting capability 4 have the same motor revision

https://community.home-assistant.io/t/adding-tilt-to-the-hunter-douglas-powerview-cover-integration-luxaflex/233720/98?u=kingy444
```json
{
        "shadeIds": [3433, 58444, 58722, 731],
        "shadeData": [{
                "id": 3433,
                "type": 55,
                "capabilities": 3,
                "batteryKind": 1,
                "smartPowerSupply": {
                    "status": 0,
                    "id": 0,
                    "port": 0
                }
            }, {
                "id": 58444,
                "type": 54,
                "capabilities": 3,
                "batteryKind": 1,
                "smartPowerSupply": {
                    "status": 0,
                    "id": 0,
                    "port": 0
                }
            }, {
                "id": 58722,
                "type": 55,
                "capabilities": 4,
                "batteryKind": 1,
                "smartPowerSupply": {
                    "status": 0,
                    "id": 0,
                    "port": 0
                }
            }, {
                "id": 731,
                "type": 54,
                "capabilities": 4,
                "batteryKind": 1,
                "smartPowerSupply": {
                    "status": 0,
                    "id": 0,
                    "port": 0
                }
            }
        ]
    }
```

**Comment by andrewfg** [comment]
^
@kingy444 many thanks for the information. But unfortunately I think your conclusion is not right because if you examine the `positions` element in the JSON posted by the user (link below) you can see that both capabilities:3 and capabilities:4 variants contain double position values, which means that both variants of the same shade type do actually support both a main and a vane position. Meaning that the "correct" value is probably capabilities:3 rather than capabilities:4. Or ???

```
"positions": {
    "posKind1": 1,
    "position1": 0,
    "posKind2": 3,
    "position2": 30849
}
```

https://community.home-assistant.io/t/adding-tilt-to-the-hunter-douglas-powerview-cover-integration-luxaflex/233720/98?u=kingy444

**Comment by andrewfg** [comment]
@jlaur could you please give me your thoughts about how we should handle these type 54/55/56 shades, where some instances report capabilities:3 and other instances capabilities:4 ???

**Comment by kingy444** [comment]
@andrewfg  I was suggesting that capability 3 is tilt 180 and capability 4 is plain open close as we have them currently and that type 54,55 & 56 should be forced to use capability 3 to get a tilt functionality regardless of what capability they return in the json. (this is how HA and this API function atm)

Really we need JSON for either 69,70 or 71 to confirm what capability 4 looks like on a different shade type.

Essentially i am saying for these shade types we now cannot trust the capability value and need to force the capability type

**Comment by jlaur** [comment]
> could you please give me your thoughts about how we should handle these type 54/55/56 shades, where some instances report capabilities:3 and other instances capabilities:4 ???

This is our current order of precedence:
- capabilitiesOverride for JSON type
- JSON capabilities
- capabilities for JSON type

So if some of these shades actually report wrong capabilities, and they all actually share the same capabilities, we should simply use capabilitiesOverride like we did for type 44 (Twist).

If actual capabilities are different from shade to shade for same type, we have to make capabilities configurable as we have no reliable way of determining correct one.

Last possibility, if the reported capabilities are correct, it will already override the type capabilities, so it should work (although we currently log a warning in this case). But from @kingy444's last comment this doesn't seem to be the case.

**Comment by andrewfg** [comment]
@jlaur @kingy444 many thanks for your inputs; I will make a new PR for the OH code.

**Comment by kingy444** [comment]
This is just an FYI on this API

V2 which will be packages soon has two methods to identify a shade. 

First via hard coded shade “type” - this makes sure shades are given the type we want them to and this is passed to HA

Second (this is the new part) does a smart check via capability and assigns an educated guess to the type of shade it should be using.
Meaning this API should in theory now support new shades as long as their capabilities are reported correctly without any API changes.

the only shortfall will be we don’t have any friendly name for them in that case but functionally they should be fine

**Comment by andrewfg** [comment]
> I was suggesting that capability 3 is tilt 180 and capability 4 is plain open close as we have them currently and that type 54,55 & 56 should be forced to use capability 3 to get a tilt functionality regardless of what capability they return in the json. (this is how HA and this API function atm)

@kingy444 picking up from the thread on the HA forum concerning capabilities 3 vs 4.
1 We had wrongly assumed that caps:3 was with tilt and caps:4 without
2. We saw a user with shade types 54,55,56 reporting caps:4 
3. The same user also had shade types 54,55,56 reporting caps:3 
4. So we applied an artificial override on those shade types to force all of them to caps:3
5. We now see that assumption 1. was wrong, and that caps:3 is without tilt and caps:4 with
6. Ergo we should to BOTH of the following..
- reverse the caps:3 vs. caps:4 definitions AND 
- continue to force override shade types 54,55,56 but change the override from caps3: to caps:4 instead

=> @kingy444 do you agree with that?
=> @jlaur pinging you on this conversation!

![image](https://user-images.githubusercontent.com/893994/187206354-cf7a2fe8-8aa4-4fbe-a9f2-20725d20bab1.png)

**Comment by andrewfg** [comment]
^
... and BTW shades type 69/70/71 (curtain) change from caps:4 to caps:3 ..

**Comment by kingy444** [comment]
@andrewfg just added 26/27/28 type definitions - Skyline Panels with capability 3
```
        shade_type(26, "Skyline Panel, Left Stack"),
        shade_type(27, "Skyline Panel, Right Stack"),
        shade_type(28, "Skyline Panel, Split Stack"),
```

**Comment by jeffdeal** [comment]
Good day folks, I have installed ARC and have successfully sent commands to my shades, type 23 and 65.  I am happy to run any tests you need.

I've done a little playing around with my shades this morning.  For my testing I have only sent values in posKind1 and position1.  I did not include posKind2 and position2 in the JSON at all.

For the type 23s:

When posKind1 = 1, sending a value in position1 between 0 and 65535 moves the shade, with 0 being fully down (closed) and 65535 being fully open (up).

Sending posKind1 = 3 and a value between 0 and 32767 will lower the shade regardless of current position and tilt the vanes.  0 is closed, 32767 is open 90 degrees.

For the type 65s:

When sending posKind1 = 2, this refers to the back panel.  Sending posKind1 = 2 and position1 between 0 and 65535 will place the back panel in the requested position, with 0 being completely closed (down) and 65535 being up (open).  Since the back and front panels cannot operate independently, the front panel will always close when commanding the back panel.  For example, if the front panel is open (up) and you send posKind1 = 2 and position1 = 65535, the front panel will close and the back will be fully open.  This is equivalent to sending posKind1 = 1 and position1 = 0.

Sending posKind1 = 1 always refers to the front panel, with position values of 0 representing down (closed) and 65535 up (open).  Since the shades cannot operate independently, the back panel always has to fully raise to achieve the desired setting.

To me, the simplest way to represent this would be a toggle switch to select the front or back panel (which would set the value of posKind1) and then a slider to represent 0 to 100% to represent the position.  Of course, I don't know if the HA architecture would permit this type of interface.

For these shades, I don't think you should filter for impossible combinations.  Just send the command and allow the shade to do what it needs to do to achieve the result.  For example, on the 23s it is impossible to tilt unless the shade is fully down.  A command to tilt the shade shouldn't be filtered, just send the tilt position and the shade knows to close first.

I'll be glad to help in any way I can, so let me know if you need any special testing or other information.

Jeff



________________________________

**Comment by kingy444** [comment]
@jeffdeal i think we have all types sorted

over on the HA forum a custom version of the component is posted to test

the only thing reported there was the open close buttons not working for type 8/9/10 (oversight in me transferring code to new format)

you’re welcome to install and test that - I’ve been working on the PR to get these into core HA this week - hopefully I can get the PR in time for 2022.11

**Comment by andrewfg** [comment]
> added 26/27/28 type

Many thanks @kingy444

**Comment by kingy444** [comment]
@andrewfg how far along are you with the implementation of the hen 3 hub?

I know some of the naming changed such as poskind1 / tilt / primary etc but do the actual endpoints still take the same json format (using new names ?)

just getting this to where it needed to be for complete gen2 support now and I’ll start v3 soon - need to work out how much I need to do

**Comment by andrewfg** [comment]
> how far along are you with the implementation of the hen 3 hub?

Hi @kingy444 .. a few updates..

- The coding is essentially finished :) -- see [PR #13355](https://github.com/openhab/openhab-addons/pull/13355)
- However we are awaiting some final information from @vves concerning last minute API changes.
- And also it is not tested; we are hoping that @vves can grant us VPN access to one of his hubs for testing.
- And my 'partner in crime' @jlaur has not yet had a chance to look at it either.

**Comment by andrewfg** [comment]
> I know some of the naming changed such as poskind1 / tilt / primary etc but do the actual endpoints still take the same json format (using new names ?)

@kingy444 .. it is 'all of the above' see this link (you can expand the buttons) https://www.tdrs.info/Swagger%20UI.html

**Comment by kingy444** [comment]
> * And also it is not tested; we are hoping that @vves can grant us VPN access to one of his hubs for testing.

Fingers crossed - i got a lot of work to get gen 3 into HA so that would be amazing

Its not clear from the api docs - do you know how shades that require both attributes to be sent behave in gen 3 ? ie in Gen2 - if you didnt post both positions the shade wouldnt return the data until a force refresh occurred

**Comment by kingy444** [comment]
also another set of capability 0 added
        shade_type(10, "Duette and Applause SkyLift"),

**Comment by kingy444** [comment]
@andrewfg did you have contact with the HD rep still / the swagger UI is no longer active

**Comment by andrewfg** [comment]
> did you have contact with the HD rep still

Yes I and @jlaur have been in email contact with him. He promised to share the final spec with us. But apparently there is still some work to be done on it, and they have to push some new firmware, so it seems to be taking him a bit longer than he originally expected. But I will keep you posted.

> the swagger UI is no longer active

I guess you are referring to the API that was posted / hosted by another HA user? Yes I also noticed that he had taken it down. For me it’s not a big issue since I think I already have enough info to work on until I get the final version from the HD guy. But if you need it then perhaps contact the OP on the HA forum and ask if he can put it up again?

**Comment by jpearl** [comment]
Heya folks - 
Just had to re-add 2 sets of shades on top of the latest version of this library in HA, and Bottom Up shades that were previously advertised with the correct capabilities (when added on an older version of HA) are now using the wrong capabilities. They only support bottom up, but in HA, they're showing a tilt capability as well (which is incorrect).

Shade type = 52
They're a simple Bottom Up shade [Banded Shades](https://www.hunterdouglas.com/window-treatments/shades/roller-shades/designer-banded)
When I dump the HA diagnostic info for them though (and hitting the hub directly), it shows `capabilities: 1`, yet only includes `posKind1` and `position`.

So are these being advertised incorrectly with capability 1 instead of capability 0? Should we just "fix" this here and add the mapping from type 52 to capability 1?

**Comment by andrewfg** [comment]
> they're showing a tilt capability as well (which is incorrect).

I think that on such banded shades, the ‘tilt’ capability actually does exist. You just need to interpret ‘tilt’ as ‘transparency’, so when you set tilt=0 the bands adjust to let the maximum light through, whereas when you set tilt=32k bands adjust to let the minimum light through. This is certainly the case on type 44 Twist shades..

**Comment by jpearl** [comment]
> I think that on such banded shades, the ‘tilt’ capability actually does exist. You just need to interpret ‘tilt’ as ‘transparency’, so when you set tilt=0% the bands adjust to let the maximum light through, whereas when you set tilt=100% bands adjust to let the minimum light through. This is certainly the case on type 44 Twist shades..

Thanks; i've tested and it's not the case, in reality, on the banded shades.

The Shade Status might **imply** that it's doing this, but it's not physical functionality on the shade. When the shade is set to closed and you adjust the tilt, the shade status does say that `posKind1` changes to `3` from `1`, but the physical shade does not move/change. 

And of course this is a real pain, as exposing this entity to a voice assistant, the VA needs to now disambiguate the setting:
 > "Set shade to 50%"
 < "Which setting do you want to adjust"

There's a super hacky workaround in HA which is to manually create a new Template Cover that only adjusts the `current_position` attribute of the shade, (since the 2 positions are not exposed as separate entities), but that's... super hacky

**Comment by kingy444** [comment]
> So are these being advertised incorrectly with capability 1 instead of capability 0? Should we just "fix" this here and add the mapping from type 52 to capability 1?

@jpearl from your description this sounds like this is what will be needed. Unfortunately this is working by design and we can’t trust Hunter Douglas are reporting type correctly. Another user with type 52 could very well have JSON reporting capability 0 as it should.

The code in this API will attempt to map via the hard coded shade type (52) and if that fails (and it will as we don’t have these defined) it will fall back to the capability (1). This was added in version 2 and works great when the blind firmware is working correctly but we have had this issue pop up where the shades think they have capabilities they don’t.

Please still post the JSON though at the end of the day we really need to take user advice on each model functionality and work from there as Hunter Douglas don’t provide this info.

**Comment by jpearl** [comment]
Sorry for the delay, here it is:

When closed fully:
```
{
  "shade": {
    "id": 58151,
    "type": 52,
    "capabilities": 1,
    "batteryKind": 2,
    "smartPowerSupply": {
      "status": 0,
      "id": 0,
      "port": 0
    },
    "batteryStatus": 0,
    "batteryStrength": 0,
    "roomId": 920,
    "firmware": {
      "revision": 2,
      "subRevision": 2,
      "build": 2754,
      "index": 32
    },
    "name": "[REDACTED]",
    "groupId": 33076,
    "positions": {
      "posKind1": 1,
      "position1": 0
    }
  }
}
```


Half open (only posting positions for brevity):
```
"positions":{"position1":32767,"posKind1":1}
```


Fully open:
```
"positions":{"position1":65535,"posKind1":1}
```

**Comment by jpearl** [comment]
@kingy444 - thoughts^?

**Comment by kingy444** [comment]
Yea it’s as I suspected and the shades are reporting capability 1 incorrectly.

as we don’t have them defined, they fallback to capability matching and give tilt based on that

I am working on an the active PR submission to support v3 atm - I have included those shades there.
